src/ui/sourceMap.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+export interface FileMetaData {
+	fileName: string;
+	filePath: string;
+	crateName: string;
+	cratePath: string;
+	harnesses: HarnessMetadata[];
+}
+
+export interface HarnessMetadata {
+	name: string;
+	fullLine: string;
+	endPosition: Position;
+	attributes: string[];
+	args: AttributeMetaData;
+}
+
+export interface AttributeMetaData {
+	/// Whether the harness has been annotated with proof.
+	proof: boolean;
+	/// Whethere the harness has been annotated with test (bolero cases)
+	test: boolean;
+	/// Optional data to store solver.
+	solver?: string;
+	/// Optional data to store unwind value.
+	unwind_value?: number;
+	/// The stubs used in this harness.
+	stubs?: Stub[];
+}
```

Comment:
I have one soundness/code smell comment and one more general design concern.

This seems to be enumerating the various attributes that Kani may currently be annotated with. The rest of the code then acts on these attributes by passing the right flags, etc. This will go out of date when Kani starts to support new attributes. This would be a problem if the correctness of the proof depends on whether the attribute is set, since this current code would ignore attributes that are not enumerated here. This is especially a problem because Kani and the extension get updated through two different mechanisms and there's no guarantee that users would keep the extension up to date.

Also, it's not clear that any of this processing needs to happen at all. I think the extension should simply invoke `cargo kani` with the name of the file. The kani driver is already responsible for reading and interpreting these attributes. So I suspect that a lot of this parsing and subsequent handling code should be removed.

We discussed offline that there is at least one attribute that has an effect on the command line: `#test`. I'm still not comfortable about the prospect that Kani will add an annotation, which like `#test` requires users to use a different command line. If this happens, the command line behavior will diverge from the extension. I'd like to be convinced that `#test` is the only attribute that makes it necessary to modify the command line.

Reply:
I have removed the attributes from the parsing module. I'll clean up the rest of the execution path in another PR. (This PR is already messy). Now attributes are just added to a list but they are not processed in anyway, and the parsing and logic is both left to `kani_driver`.

Reply:
Thank you!